### PR TITLE
Add Makefile support for add variables overrides to a .env file

### PR DESCRIPTION
**Pull request by LaVLaS**
## Description
Adds support for overriding Makefile variablies in an `.env` file.  This will allow developers to store custom configurations in a file without having to pass the override to `make` on the command line

I removed IMAGE_OWNER  based on the expectation that the user can now use the .env file to set `IMAGE_TAG_BASE`  value directly.

## How Has This Been Tested?
* Manually tested with overrides for common variables in the Makefile: VERSION, IMG, IMAGE_TAG_BASE, IMAGE_TAG, E2E_TEST_FLAGS, IMAGE_BUILD_FLAGS

## Merge criteria:
* Makefile execution is unaffected when the `.env` file does not exist
* No builds in openshift-ci are broken by this change
* Other developers can customize `.env` for their local environment and run `make <TARGET>` as intended 

## Additional Info
I did not make any changes to default variable assignments (other than IMAGE_OWNER) based on the expectation that existing functionality should be maintained

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by ykaliuta** [comment]
I would rather call it something like `local.mk` or `include.mk` than `.env`. The term `environment` has some sense here for sure, but it has precise meaning in the OS, the same applies for setting environment for containers. It could fit more if the file could be used as a shell include as well, but in your example it is not written such a way. Having it hidden makes the overriding less obvious.

About `?=` see the original PR (but to make it work properly all the overridable variables should be moved before the include)
 
But of course I'll not argue :)

File: Makefile [hunk 1]
```diff
@@ -93,7 +88,16 @@ E2E_TEST_FLAGS = "--skip-deletion=false" -timeout 15m # See README.md, default g
...
-IMAGE_BUILD_FLAGS = --build-arg USE_LOCAL=false
+IMAGE_BUILD_FLAGS ?= --build-arg USE_LOCAL=false
+
+# Read any custom variables overrides from a .env file.  This will only be read if it exists in the 
+# same directory as this Makefile.  Variables can be specified in the standard format supported by 
+# GNU Make since include process Makefiles
+# Standard variables override would include anything you would pass at runtime that is different
+# from the defaults specified in this file
+OPERATOR_MAKE_ENV_FILE = .env
+-include $(OPERATOR_MAKE_ENV_FILE)
```
Comment (zdtsw) [inline Makefile:99]: should we move this block in the top of Makefile?

Comment (ykaliuta) [reply Makefile:99]: I would say no. I would vote for the idea to have variables defaults set above it and then override them in the include. Then get rid of all the '?=' to ensure no accidental initialization from process environment.

Comment (ykaliuta) [reply Makefile:99]: I just do not see any sense of introducing a variable here. Is it used somewhere else? If you want emphasize that it is Makefile, may be it is enough to add .mk suffix.

Comment (LaVLaS) [reply Makefile:99]: That is a good question. Originally, I had it at the top but that would restrict overrides to any variable below that utilizes `VAR ?= VALUE`.  If there is no real value in putting it at the bottom then I have no issues moving it back to the top for visibility

**Comment by ykaliuta** [comment]
> 
> I have no issues with the use of `?=` for variable assignment as I think the default functionality should the ability to run `make VAR=VALUE <target>` as without requiring this functionality

Ah, don't worry, make VAR=VALUE works with = fine. Command line is the top priority for make variables.

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**